### DISCUSSION TITLE
Parameter tab. Do I use it correctly?

### DISCUSSION THREAD

**Discussion by miracuruzrh** [Part-DB/Part-DB-server#640] [Q&A] [answered]
Hi

First of all. I love this project. Many thanks for this great repository.
I have tested out a few different inventory databases and decided to go with this, because it's relatively simple, clear structured and easy to use.

I'm building my inventory from scratch. So I'm not yet sure, how to structure my data. Therefore I like to ask early in the beginning, so that I don't go into a wrong direction and build up a wrong structure.

I'm from Switzerland, therefore I have some datasheets in German and some in English. Now I'm not sure about the parameter tab and what kind of data I should fill in. It seems like that the parameter fields are a relatively loose structure. For example, for a parameter for Forward current [If]. This would be Durchlass Spannung [If] in German. Let's say, I fill out this field for one part in German and for other parts in English. Then I guess, my Data is not really searchable for this parameter in the future, when I have mixed data. Or let's say some parts uses frequency and other uses wavelength and so on.

Therefore I'm not 100% sure, if this parameter field tab is really correct for this kind of data. Wouldn't it be more clearer, if this fields would be like the vendor or supplier fields, where you have a very defined structure. Or like categories, where you first define the categories, and only the you are able to use the category. Then It would be also possible to define for example frequency and also define a alias, called wavelenght when it's interchangeable for frequency and so on.

Would be nice to get a few opinions on this and to see if I maybe misunderstood the concept of parameters.

Here is a example of a LED I've entered into my database.
![image](https://github.com/Part-DB/Part-DB-server/assets/44341190/00229c59-855b-444e-9c39-abc43d279856)

**Comment by jbtronics** [comment, marked as answer]
You are right. The parameters can be defined very freely and have almost no pregiven structure. But that is more or less intentional.

For what kind of data you wanna use the parameters, is your decision. To be useful you should have a concept for it, where you use the same names and units for all your parts (however Part-DB should make you suggestions based on existing parameters, so that is not too difficult).

>  Wouldn't it be more clearer, if this fields would be like the vendor or supplier fields, where you have a very defined structure.

You can not know what kind of parts users will put into their database, and the only common structure is that you have some kind of parameter with a value range, a typical value and or a text value. And that is bascially what Part-DBs parameter system allows you to define.

> Or like categories, where you first define the categories, and only the you are able to use the category. 

The problem is that not every part shares the same set of parameter values, even in the same category.
For some parts you might not know a certain parameter, and others have some additional parameter, which might be nice to input into the database but is not known for the other parts. 

So basically these category defined parameters could just be suggestions, otherwise the system becomes pretty unflexible.
And Part-DB already makes you suggestions for parameters, based on existing ones.

**Reply by miracuruzrh** [reply]
Dear @jbtronics 

Thanks for your reply.
Okay I get it. I had meanwhile also al look into the demo Part-DB database, how there the parameters are organized. And I think I understand the concept now better. I have also connected my Part-DB Instance to Mouser and DigiKey as well and that make createing new parts much faster, as all the data and parameters get pulled into my database from Mouser and Digikey.

At first, I had a more or less clear concept in mind, which parameters I want to enter in my parts. But I think now, that there came already so many parameters with the API connection to Digikey, that my concept is obsolete now. Because I have now already all kind of different suggestions like Pin, or Pins and Row and Rows and Pin per rows and so on.

I think I just enter, what makes the most sense when I have to fill in parameters by myself and when it comes to the point, I need to search for this parameters. I just search with the generic search field and not with the detailed filter field, where I can search for a exactly defined parameter.
